main.js

```diff
@@ -62,9 +60,9 @@ class Covid19 extends utils.Adapter {
 				await this.addUserCountriesTranslator();
 
 				// Write all country states depending on filter
-				for (const i in values) {
-					if (values.hasOwnProperty(i) && values[i] && values[i].country) {
-						let country = values[i].country;
+				for (const dataset of values) {
+					if (dataset.country) {
+						let country = dataset.country;
```

Comment:
values is an array. By iterating with `of` you can be sure that all values exist. Also I like speaking names, so `values[i] ==> dataset`

Reply:
nice one thanks !